parquet-hadoop/src/main/java/org/apache/parquet/crypto/CryptoPropertiesFactory.java

```diff
@@ -51,20 +71,36 @@ public static CryptoPropertiesFactory loadFactory(Configuration conf) throws IOE
 
     try {
       CryptoPropertiesFactory cryptoFactory = (CryptoPropertiesFactory)cryptoPropertiesFactoryClass.newInstance();
-      cryptoFactory.initialize(conf);
       return cryptoFactory;
     } catch (InstantiationException | IllegalAccessException e) {
       throw new BadConfigurationException("could not instantiate CryptoPropertiesFactory class: "
         + cryptoPropertiesFactoryClass, e);
     }
   }
 
-  public abstract void initialize(Configuration hadoopConfig);
-
-  public abstract FileEncryptionProperties getFileEncryptionProperties(
-    Configuration fileHadoopConfig, Path tempFilePath,
-    WriteContext fileWriteContext)  throws IOException;
+  /**
+   * Get FileEncryptionProperties object which is created by the implementation of this interface. Please see
+   * the unit test (TBD) for example
+   *
+   * @param fileHadoopConfig Configuration that is used to pass the needed information, e.g. KMS uri
+   * @param tempFilePath File path of the parquet file being written.
+   *                     Can be used for AAD prefix creation, key material management, etc.
+   *                     Implementations must not presume the path is permanent,
+   *                     as the file can be moved or renamed later
+   * @param fileWriteContext WriteContext to provide information like schema to build the FileEncryptionProperties
+   * @return
+   * @throws IOException
+   */
+  FileEncryptionProperties getFileEncryptionProperties(Configuration fileHadoopConfig, Path tempFilePath,
+                                                       WriteContext fileWriteContext)  throws IOException;
 
-  public abstract FileDecryptionProperties getFileDecryptionProperties(
-    Configuration hadoopConfig, Path filePath)  throws IOException;
+  /**
+   *
+   * @param hadoopConfig Configuration that is used to pass the needed information, e.g. KMS uri
+   * @param filePath File path of the parquet file
+   *                 Can be used for AAD prefix verification, part of key metadata etc
+   * @return
+   * @throws IOException
+   */
```

Comment:
Missing main description, empty return and throws tags.

Reply:
Fixed